cli/azd/pkg/appservice/appservice.go

```diff
@@ -0,0 +1,114 @@
+package appservice
```

Comment:
Does this also add support for functions deployed with containers?  I would like to add that at the same time if possible.

Reply:
No, this doesn't tackle Function Apps, I wanted to scope it to one service at a time (although I'd expect it to be pretty similar for Functions)

Reply:
I would like to have them both implemented at the same time because they are so similar and I think the function features of this may drive changes, potential API changes, that we didn't account for when isolated to app service.

I also think there's a lot of common code between the two, so the code you have written for app service would be used by functions, and could be pulled out into a shared lib.

Reply:
Ok, I'll add Functions to this PR and look at how we can share code.

Reply:
Added Functions logic too. Since they share the same structure under the hood I've just repurposed the new logic for appservice into the Functions branch and used some of the same types.

Still having the problem where I'm not sure how to reference this new file so I can test it out.

Reply:
@vhvb1989 - Is this something you can help with?

Reply:
> Still having the problem where I'm not sure how to reference this new file so I can test it out.

Can you elaborate?

Reply:
@aaronpowell , it was a simple typo, you need to use `appservice.` to reference the package instead of `appService.`

I fixed it (and a few other lint issues) for you and pushed.  Also pulled latest main.

You should be able to build and test now